surefire-providers/surefire-junit-platform/src/main/java/org/apache/maven/surefire/junitplatform/JUnitPlatformProvider.java

```diff
@@ -147,7 +149,26 @@ private TestsToRun scanClasspath()
     private void invokeAllTests( TestsToRun testsToRun, RunListener runListener )
     {
         LauncherDiscoveryRequest discoveryRequest = buildLauncherDiscoveryRequest( testsToRun );
-        launcher.execute( discoveryRequest, new RunListenerAdapter( runListener ) );
+        RunListenerAdapter adapter = new RunListenerAdapter( runListener );
+        launcher.execute( discoveryRequest, adapter );
+        // Rerun failing tests if requested
+        int count = parameters.getTestRequest().getRerunFailingTestsCount();
+        if ( count > 0 && adapter.hasFailingTests() )
+        {
+            for ( int i = 0; i < count; i++ )
+            {
+                // Replace the "discoveryRequest" so that it only specifies the failing tests
+                discoveryRequest = buildLauncherDiscoveryRequestForRerunFailures( adapter );
+                // Reset adapter's recorded failures and invoke the tests again
```

Comment:
Added one word: `// Reset adapter's recorded failures and invoke the failed tests again`